docs/tutorial/working-with-frames-and-series/index.md

```diff
@@ -0,0 +1,213 @@
+------------------+-------------------+-------------------+------------------
+0                 | 1973-07-01        | female            | 2015-09-14
+1                 | 1948-03-01        | male              | None
+2                 | 2003-04-01        | male              | None
+3                 | 2007-06-01        | female            | None
+4                 | 1938-10-01        | male              | 2018-05-23
+5                 | 1994-04-01        | female            | None
+6                 | 1953-05-01        | male              | None
+7                 | 1992-08-01        | female            | None
+8                 | 1931-10-01        | female            | 2017-11-10
+9                 | 1979-04-01        | male              | None
+```
+
+### Work with a patient series
+
+The `patients.date_of_birth` column is a patient series; it has one row per patient.
+
+```pycon
+>>> patients.date_of_birth
+0 | 1973-07-01
+1 | 1948-03-01
+2 | 2003-04-01
+3 | 2007-06-01
+4 | 1938-10-01
+5 | 1994-04-01
+6 | 1953-05-01
+7 | 1992-08-01
+8 | 1931-10-01
+9 | 1979-04-01
+```
+
+## Work with event data
+
+### Import the `medications` table
+
+```pycon
+>>> from ehrql.tables.beta.core import medications
+```
+
+### Work with an event frame
+
+The `medications` table is an event frame; it has many rows per patient.
+Notice that some values in the `patient_id` column are not unique.
+
+```pycon
+>>> medications
+patient_id        | row_id            | date              | dmd_code
+------------------+-------------------+-------------------+------------------
+0                 | 0                 | 2014-01-11        | 39113611000001102
+1                 | 1                 | 2015-08-06        | 39113611000001102
+1                 | 2                 | 2018-09-21        | 39113311000001107
+1                 | 3                 | 2020-05-17        | 22777311000001105
+3                 | 4                 | 2022-11-09        | 22777311000001105
+4                 | 5                 | 2017-05-11        | 39113611000001102
+5                 | 6                 | 2017-07-11        | 3484711000001105
+5                 | 7                 | 2019-07-06        | 39113611000001102
+7                 | 8                 | 2021-01-27        | 3484711000001105
+9                 | 9                 | 2015-03-14        | 3484711000001105
+```
+
+### Work with an event series
+
+The `medications.date` column is an event series; it has many rows per patient.
+
+```pycon
+>>> medications.date
+0 | 0 | 2014-01-11
+1 | 1 | 2015-08-06
+1 | 2 | 2018-09-21
+1 | 3 | 2020-05-17
+3 | 4 | 2022-11-09
+4 | 5 | 2017-05-11
+5 | 6 | 2017-07-11
+5 | 7 | 2019-07-06
+7 | 8 | 2021-01-27
+9 | 9 | 2015-03-14
+```
+
+## Transform input data into output data
+
+**Tables contain input data**.
+As a researcher, your task is to transform tables into a dataset.
+A dataset, like a patient frame, has one row per patient.
+Unlike a patient frame, however, **a dataset contains output data**.
```

Comment:
I found this section hard to parse. Thinking it through aloud:

> **Tables contain input data**.

Me: Hm, why "input" data? Isn't it just data? Is it because we care about where it came from? 

> As a researcher, your task is to transform tables into a dataset.

Me: what is a dataset? Is it like a table? How is it different?

> A dataset, like a patient frame, has one row per patient.

Me: what's a patient frame again? [scrolls back] Oh, it's a table. Hang on, why aren't we calling it a table?

> Unlike a patient frame, however, **a dataset contains output data**.

Me: Doesn't a patient frame contain data?! Is that _input_ data? How are they different? I thought it was _tables_ that contain input data. 


I've gotta dash, sorry, but happy to try to come up with some alternative suggestions later if it helps!

Reply:
Thanks, @sebbacon. I'm unconvinced by this section; but I couldn't remove _everything_ from the existing tutorial 🙂 I think it was originally here to motivate the transformations that follow. We now have them, and many more, in [*Using ehrQL to answer specific questions*](https://docs.opensafely.org/ehrql/how-to/examples/). 

It doesn't help that "table" is overloaded. Here, we mean specifically ehrQL tables (not, for example, tabular data structures). I'll clarify.

We care very much about where the data came from: it came from L2. But in this case, it didn't, because it's dummy input data: it came from a proxy L2. As a researcher, you should know that data come from somewhere and go to somewhere. You should know that ehrQL sits in the middle.

I feel that the "What's a dataset?" question is answered: one row per patient; output data. The reference to a patient frame isn't helpful; I'll remove.

Reply:
I think I'd also like to add that to a certain extent, the names are given: Whilst we may decide that we could tighten up "table", "frame", "input data", and "output data", I've tried to update the tutorial to reflect what we have now, rather than what we might have in the future.